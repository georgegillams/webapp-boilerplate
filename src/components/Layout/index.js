import React from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';
import appConfig from 'helpers/appConfig';

import Header from 'components/Header';
import Banner from 'components/Banner';
import Footer from 'components/Footer';

function Layout({ children }) {
  return (
    <>
      <Head>
        <title>{appConfig.app.title}</title>
        {appConfig.app.head.meta.map(m => (
          <meta key={m.property} name={m.property} content={m.content} />
        ))}
      </Head>
      <Header />
      <main>
        <Banner />
        {children}
      </main>
      <Footer />
    </>
  );
}

Layout.propTypes = {
  children: PropTypes.node,
};

export default Layout;
