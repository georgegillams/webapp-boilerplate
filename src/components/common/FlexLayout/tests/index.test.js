import React from 'react';

import { render } from '@testing-library/react';

import Layout from '..';

describe('<Layout />', () => {
  const children = <div>some text</div>;

  it('Should render correctly', () => {
    global.Date.getMonth = jest.fn(() => 2);
    const { container } = render(<Layout>{children}</Layout>);

    expect(container).toMatchSnapshot();
  });
});
