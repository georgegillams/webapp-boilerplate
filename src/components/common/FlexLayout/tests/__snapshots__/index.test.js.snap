// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Should render correctly 1`] = `
<div>
  <main
    class="flex-layoutstyles__StyledMainWrapper-sc-18pl5kb-0 bvXxXW"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;
